packages/core/src/collections/notes.ts

```diff
@@ -121,9 +125,33 @@ export class Notes implements ICollection {
             this.db.settings.getTitleFormat(),
             this.db.settings.getDateFormat(),
             this.db.settings.getTimeFormat(),
-            headline?.split(" ").splice(0, 10).join(" ") || "",
+            headline ? headlineToTitle(headline) : "",
             this.totalNotes
           );
+        item.isGeneratedTitle = true;
+      }
+
+      const currentNote = await this.note(id);
```

Comment:
We don't need the whole note. How about using a custom query to only get the fields we need?